git_trunk/tests/test_git_trunk_refresh.py

```diff
@@ -78,3 +78,15 @@ def test_02_git_refresh(self):
         self.git.add('.')
         self.git.commit('-m', 'new commit blabla')
         self._test_git_refresh('branch1', diff='')
+
+    def test_03_git_refresh(self):
+        """
+        Refresh trunk branch when having changes on submodule.
+        """
+        self._create_dummy_submodule('init_submodule_dir')
+        self._update_dummy_file_in_dir('init_submodule_dir')
+        trunk_refresh = GitTrunkRefresh(
+            repo_path=self.dir_local.name,
+            log_level=common.LOG_LEVEL,
+        )
+        trunk_refresh.run()
```

Comment:
But where is actual test? You run refresh, but you do not check if refresh was done correctly. It could be the fact that refresh fails silently and does not do refresh correctly.

If you would check other refresh tests, you would see there is helper `_test_git_refresh`. You could use it too, to make sure refresh does what its supposed to do.

Reply:
I added a diff check before and after refresh, because once submodule is added diff is dirty and its keeps dirty after refresh is completed, you think its good or still would need to improve it?

Reply:
It doesnt matter its dirty or not. Diff should still be the same before refresh and after (cause you use plain diff check in both cases).